### ISSUE TITLE
Unable to serve pre-existing files anymore

### ISSUE THREAD

**Issue by leshik** [minio/minio#5893]
## Expected Behavior

Pre-existing files are serving successfully, e.g.:

```
> curl http://localhost:9000/test/empty.json
{}
```

## Current Behavior

Internal Server Error (HTTP code 500) and `minio` complains:

```
API: GetObject(bucket=test, object=empty.json)
Time: 03:56:04 UTC 05/07/2018
RemoteHost: 172.17.0.1:45234
UserAgent: curl/7.59.0
Error: etag of the object has changed
       1: cmd/fs-v1.go:489:cmd.(*FSObjects).getObject()
       2: cmd/fs-v1.go:443:cmd.(*FSObjects).GetObject()
       3: cmd/object-handlers.go:175:cmd.(ObjectLayer).GetObject-fm()
       4: cmd/object-handlers.go:181:cmd.objectAPIHandlers.GetObjectHandler()
       5: cmd/api-router.go:73:cmd.(objectAPIHandlers).GetObjectHandler-fm()
       6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()
```

## Steps to Reproduce

Just use `minio` version `RELEASE.2018-04-27T23-33-52Z` or `RELEASE.2018-05-04T23-13-12Z`.

## Context

I'm unable to serve pre-existing files with `minio` anymore. Everything was perfect with `RELEASE.2018-04-04T05-20-54Z`, but it doesn't work with `RELEASE.2018-04-27T23-33-52Z` and `RELEASE.2018-05-04T23-13-12Z`.

Starting from `RELEASE.2018-04-12T23-41-09Z` minio throws this error eight times to the console when starting:

```
API: SYSTEM()
Time: 03:48:42 UTC 05/07/2018
Error: volume access denied
       1: cmd/fs-v1-helpers.go:186:cmd.fsStatVolume()
       2: cmd/fs-v1.go:280:cmd.(*FSObjects).ListBuckets()
       3: cmd/bucket-policy.go:86:cmd.initBucketPolicies()
       4: cmd/fs-v1.go:143:cmd.NewFSObjectLayer()
       5: cmd/server-main.go:300:cmd.newObjectLayer()
       6: cmd/server-main.go:271:cmd.serverMain()
```

and this one eight times when trying to get the list of objects using `mc`:

```
API: ListBuckets()
Time: 03:49:57 UTC 05/07/2018
RemoteHost: 172.17.0.1:45176
UserAgent: Minio (darwin; amd64) minio-go/6.0.1 mc/2018-04-28T00:08:20Z
Error: volume access denied
       1: cmd/fs-v1-helpers.go:186:cmd.fsStatVolume()
       2: cmd/fs-v1.go:280:cmd.(*FSObjects).ListBuckets()
       3: cmd/bucket-handlers.go:226:cmd.(ObjectLayer).ListBuckets-fm()
       4: cmd/bucket-handlers.go:242:cmd.objectAPIHandlers.ListBucketsHandler()
       5: cmd/api-router.go:117:cmd.(objectAPIHandlers).ListBucketsHandler-fm()
       6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()
```

but files were served ok. Then starting from `RELEASE.2018-04-27T23-33-52Z` it stopped serving files with:

```
API: GetObject(bucket=test, object=empty.json)
Time: 03:54:10 UTC 05/07/2018
RemoteHost: 172.17.0.1:45212
UserAgent: curl/7.59.0
Error: etag of the object has changed
       1: cmd/fs-v1.go:489:cmd.(*FSObjects).getObject()
       2: cmd/fs-v1.go:443:cmd.(*FSObjects).GetObject()
       3: cmd/object-handlers.go:174:cmd.(ObjectLayer).GetObject-fm()
       4: cmd/object-handlers.go:180:cmd.objectAPIHandlers.GetObjectHandler()
       5: cmd/api-router.go:73:cmd.(objectAPIHandlers).GetObjectHandler-fm()
       6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()

```

Bucket policies are set to `download`.

## Your Environment

* Version used (`minio version`): `RELEASE.2018-04-27T23-33-52Z` or `RELEASE.2018-05-04T23-13-12Z`
* Environment name and version (e.g. nginx 1.9.1): Docker or native version – doesn't matter
* Operating System and version (`uname -a`): macOS 10.13.14, Ubuntu 16.04, Docker 18.03.1-ce – doesn't matter

**Comment by nitisht** [comment]
This was fixed in https://github.com/minio/minio/pull/5889 , will be merged soon @leshik

**Comment by leshik** [comment]
@nitisht Great! And what about these weird `volume access denied` errors?

**Comment by harshavardhana** [comment]
> API: ListBuckets()
> Time: 03:49:57 UTC 05/07/2018
> RemoteHost: 172.17.0.1:45176
> UserAgent: Minio (darwin; amd64) minio-go/6.0.1 mc/2018-04-28T00:08:20Z
> Error: volume access denied
>        1: cmd/fs-v1-helpers.go:186:cmd.fsStatVolume()
>        2: cmd/fs-v1.go:280:cmd.(*FSObjects).ListBuckets()
>        3: cmd/bucket-handlers.go:226:cmd.(ObjectLayer).ListBuckets-fm()
>        4: cmd/bucket-handlers.go:242:cmd.objectAPIHandlers.ListBucketsHandler()
>        5: cmd/api-router.go:117:cmd.(objectAPIHandlers).ListBucketsHandler-fm()
>        6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()

This is because `ListBuckets` call is trying to ignore the files that are on top along with the directories (buckets), but it logs it anyways since it has found files which are not directories.  Do you have files along with directories?

**Comment by leshik** [comment]
@harshavardhana Oh, I see. Yes, there were files on top along with the buckets, now I tested with clean top with only directories and errors are gone, thank you!

**Comment by harshavardhana** [comment]
Fixed by #5889
